### ISSUE TITLE
SSDP scan uses "ssdp:all" instead of "urn:Belkin:service:basicevent:1"

### ISSUE THREAD

**Issue by meatheadmike** [pywemo/pywemo#112]
SSDP scan basically tells every single upnp device on the network to respond. I can't figure out why this is needed. It certainly adds a lot of unwanted network traffic from what I'm seeing on my end (router, chromecasts, harmony devices, etc). From what I can determine all wemo devices respond with "urn:Belkin:service:basicevent:1". So we should be able to narrow this down significantly. If you're OK with it I can provide a PR. Let me know what you think...

**Comment by pavoni** [comment]
If this works it sounds like a win with no disadvantages (and in fact may help with the odd problem where we've not understood the reply from a non wemo device and crashed!).

So a PR would be very welcome.

**Comment by sqldiablo** [comment]
I'll bake this into my rediscovery fixes and test it along with those changes. I've been annoyed by this as well, so thanks for suggesting the fix for it. :-)

**Comment by meatheadmike** [comment]
Something else to consider while you're working on the big fix - I did a wireshark packet cap on the SSDP traffic to see what what going on during discovery. It turns out the official app actually sends out 2 separate M-SEARCH requests in fairly quick succession once it opens. This kind of makes sense since it's UDP and thus by definition has no delivery guarantee. In practice pywemo tends to under-discover my devices. This is likely due in part to my saturated network. Implementing 2 M-SEARCH requests like the official app seems to help.

**Comment by sqldiablo** [comment]
It turns out that the older WeMo devices (like the WeMo switch, the older, more oval-shaped smart plug) don't respond to the URN search (I tried), and instead only respond to a "upnp:rootdevice" search. I can take out the "ssdp:all" search and replace it with the URN search, but pywemo is using the "upnp:rootdevice" search to find devices, and isn't even sending the "ssdp:all" search during normal discovery.

The fun part is that HA still discovered the device when it ran a discovery scan using its built-in discovery component and doing a netdisco discovery, even when pywemo's discovery scan didn't find the device. When this happened, the device was still added to Home Assistant successfully and worked correctly. But since that is Home Assistant functionality, we can't rely on that in the base pywemo module, so I have to leave it with the "upnp:rootdevice" search.

**Comment by sqldiablo** [comment]
I also tried the double scan, but it gave me back duplicate results, which I don't like. I'm going to leave the second scan out of things for now.

**Comment by meatheadmike** [comment]
Interesting. I'm was able to see my wemo switches (I have 2 of the older style). I did have to replace both "upnp:rootdevice" and "ssdp:all" though. The official app only seems to use "urn:Belkin:service:basicevent:1" from what I was able to see on the packet capture. I'll play around a bit more with it and verify. 

As for the 2 separate M-SEARCH requests, I can see where you are coming from there. Still, it would be nice to have an option for those of us who have busy networks. Not a deal breaker though.
